### PR TITLE
Add ability to restrict API to authenticated only

### LINKED ISSUE #1879: Allow completely restricting /api/ endpoints to those with authorization
For the internal mozilla use case, we would like to be able to restrict *any* access to /api/ to those clients with either a token or a valid session. Currently we rely on openidc auth for these endpoints but this approach doesn't work well with the ajax approach we use inside the notebook.

For this to be useful, we need #1755 first.

/cc @robotblake

### PR DESCRIPTION
**Pull request by jezdez**
Fix #1879.

### Pull Request checklist

- [ ] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](../CHANGELOG.md) with any user-visible changes.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

**Comment by jezdez** [comment]
Tests upcoming... 😬

File: server/tests/test_jwt_authentication.py [hunk 1]
```diff
@@ -1,6 +1,3 @@
-from datetime import timedelta
-from time import sleep
```
Comment (wlach) [inline server/tests/test_jwt_authentication.py]: We should be running flake8 to detect these errors

**Comment by jezdez** [comment]
> There seems to be a black linter error. Aside from that, looks good to me!

Yeah, there is https://github.com/timothycrosley/isort/issues/694 that documents a incompatibility of isort and black in their default configuration. Luckily there is a [workaround](https://black.readthedocs.io/en/stable/the_black_code_style.html?highlight=isort#how-black-wraps-lines).

File: server/tests/test_jwt_authentication.py (hunk 1 above)
Comment (jezdez) [reply server/tests/test_jwt_authentication.py]: Done in #1931.

**Pull request merged**